### DISCUSSION TITLE
Halo2学术贡献？

### DISCUSSION THREAD

**Discussion by kurtpan666** [z2o-k7e/halo2-step-by-step#5] [Q&A]
除去工程优化，Halo2（ 相比于Halo）有哪些点是可以抽象为学术贡献的？
对后续方案有启发的点也行。
比如，在arithmetisation方面有什么贡献？在accumulation scheme方面有什么贡献？在lookup argument方面有什么贡献？

**Comment by qizhou** [comment]
Halo不太熟悉，Halo也是写plonkish电路的吗？

**Reply by kurtpan666** [reply]
Halo is a paper [BCG19] about a new (recursive) zk-SNARK proof system vs Halo2 is an optimized zk-SNARK Rust implementation of Halo (with new ideas incorporated into).
So basically you can only fork Halo2 on Github. If you want to "fork" Halo, you need to write a new paper, like [BCMS20].

Ref:
- [Halo: Recursive Proof Composition without a Trusted Setup](https://electriccoin.co/blog/halo-recursive-proof-composition-without-a-trusted-setup/)
- [Explaining Halo 2](https://electriccoin.co/blog/explaining-halo-2/)
- \[BCMS20\]: https://eprint.iacr.org/2020/499
